Unused variables in test_multiple_pvc_concurrent_creation_deletion

Test `test_multiple_pvc_concurrent_creation_deletion` defines variables `pvc_base_name` and `pvc_base_name_new` but these variables are not used for pvc creation but they are used in defined delete command:
```
         # Start deleting 100 PVCs                                                                                                  
         command = (                                                                                                                
             f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '                                                             
             f'{self.pvc_base_name}$i -n {self.namespace};done'                                                                     
         )  
```
Because of that there are no pvcs found during deletion and test fails:
```
Error Message

AssertionError: Deletion of PVCs failed assert not 1

Stacktrace

self = <tests.manage.test_pvc_concurrent_deletion_creation.TestMultiplePvcConcurrentDeletionCreation object at 0x7fd43a590400>

    def test_multiple_pvc_concurrent_creation_deletion(self):
        """
        To exercise resource creation and deletion
        """
        # Start deleting 100 PVCs
        command = (
            f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
            f'{self.pvc_base_name}$i -n {self.namespace};done'
        )
        proc = run_async(command)
        assert proc, (
            f'Failed to execute command for deleting {self.number_of_pvc} PVCs'
        )
    
        # Create 100 PVCs
        pvc_objs = create_multiple_pvcs(
            sc_name=self.sc_obj.name, namespace=self.namespace,
            number_of_pvc=self.number_of_pvc
        )
        log.info(f'Created {self.number_of_pvc} new PVCs.')
        self.pvc_objs_new = pvc_objs[:]
    
        # Verify PVCs are Bound
        for pvc in self.pvc_objs_new:
            pvc.reload()
            assert pvc.status == constants.STATUS_BOUND, (
                f'PVC {pvc.name} is not Bound'
            )
        log.info('Verified: Newly created PVCs are in Bound state.')
    
        # Verify command to delete PVCs
        ret, out, err = proc.async_communicate()
        log.info(
            f'Return values of command: {command}.\nretcode:{ret}\nstdout:'
            f'{out}\nstderr:{err}'
        )
>       assert not ret, 'Deletion of PVCs failed'
E       AssertionError: Deletion of PVCs failed
E       assert not 1

tests/manage/test_pvc_concurrent_deletion_creation.py:133: AssertionError
```

--- a/tests/manage/pv_services/test_pvc_concurrent_deletion_creation.py
+++ b/tests/manage/pv_services/test_pvc_concurrent_deletion_creation.py
@@ -2,17 +2,17 @@
 Test to verify concurrent creation and deletion of multiple PVCs
 """
 import logging
+from concurrent.futures import ThreadPoolExecutor
 import pytest
 
-from ocs_ci.ocs import constants, ocp, exceptions
-from ocs_ci.utility.utils import run_async
+from ocs_ci.ocs import exceptions
 from ocs_ci.framework.testlib import tier1, ManageTest
-from ocs_ci.ocs.resources.pod import get_ceph_tools_pod
 from ocs_ci.ocs.resources.pvc import delete_pvcs
 from tests.fixtures import (
-    create_rbd_storageclass, create_ceph_block_pool, create_rbd_secret
+    create_rbd_storageclass, create_ceph_block_pool, create_rbd_secret,
+    create_project, create_pvcs
 )
-from tests.helpers import create_unique_resource_name, create_multiple_pvcs
+from tests.helpers import create_multiple_pvcs
 
 log = logging.getLogger(__name__)
 
@@ -22,118 +22,64 @@ def test_fixture(request):
     """
     Setup and teardown
     """
-    self = request.node.cls
+    cls_ref = request.node.cls
+    cls_ref.pvc_objs_new = []
 
     def finalizer():
-        teardown(self)
+        # Delete newly created PVCs
+        assert delete_pvcs(cls_ref.pvc_objs_new), 'Failed to delete PVCs'
+        log.info(f'Newly created {cls_ref.num_of_pvcs} PVCs are now deleted.')
+
     request.addfinalizer(finalizer)
-    setup(self)
-
-
-def setup(self):
-    """
-    Create new project
-    Create PVCs
-    """
-    # Create new project
-    self.namespace = create_unique_resource_name('test', 'namespace')
-    self.project_obj = ocp.OCP(kind='Project', namespace=self.namespace)
-    assert self.project_obj.new_project(self.namespace), (
-        f'Failed to create new project {self.namespace}'
-    )
-    # Create 100 PVCs
-    pvc_objs = create_multiple_pvcs(
-        sc_name=self.sc_obj.name, namespace=self.namespace,
-        number_of_pvc=self.number_of_pvc
-    )
-    log.info(f'Created initial {self.number_of_pvc} PVCs')
-    self.pvc_objs_initial = pvc_objs[:]
-
-    # Verify PVCs are Bound and fetch PV names
-    for pvc in self.pvc_objs_initial:
-        pvc.reload()
-        assert pvc.status == constants.STATUS_BOUND, (
-            f'PVC {pvc.name} is not Bound'
-        )
-        self.initial_pvs.append(pvc.backed_pv)
-    log.info(f'Initial {self.number_of_pvc} PVCs are in Bound state')
-
-
-def teardown(self):
-    """
-    Delete PVCs
-    Delete project
-    """
-    # Delete newly created PVCs
-    assert delete_pvcs(self.pvc_objs_new), 'Failed to delete PVCs'
-    log.info(f'Newly created {self.number_of_pvc} PVCs are now deleted.')
-
-    # Switch to default project
-    ret = ocp.switch_to_default_rook_cluster_project()
-    assert ret, 'Failed to switch to default rook cluster project'
-
-    # Delete project created for the test case
-    self.project_obj.delete(resource_name=self.namespace)
 
 
 @tier1
 @pytest.mark.usefixtures(
+    create_project.__name__,
     create_rbd_secret.__name__,
     create_ceph_block_pool.__name__,
     create_rbd_storageclass.__name__,
+    create_pvcs.__name__,
     test_fixture.__name__
 )
 class TestMultiplePvcConcurrentDeletionCreation(ManageTest):
     """
     Test to verify concurrent creation and deletion of multiple PVCs
     """
-    number_of_pvc = 100
-    pvc_base_name = 'test-pvc'
-    pvc_base_name_new = 'test-pvc-re'
-    initial_pvs = []
-    pvc_objs_initial = []
-    pvc_objs_new = []
+    num_of_pvcs = 10
+    pvc_size = '3Gi'
 
     def test_multiple_pvc_concurrent_creation_deletion(self):
         """
         To exercise resource creation and deletion
         """
+        executor = ThreadPoolExecutor(max_workers=1)
+
         # Start deleting 100 PVCs
-        command = (
-            f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
-            f'{self.pvc_base_name}$i -n {self.namespace};done'
-        )
-        proc = run_async(command)
-        assert proc, (
-            f'Failed to execute command for deleting {self.number_of_pvc} PVCs'
+        log.info('Start deleting PVCs.')
+        pvc_delete = executor.submit(
+            delete_pvcs, self.pvc_objs
         )
 
         # Create 100 PVCs
-        pvc_objs = create_multiple_pvcs(
+        log.info('Start creating new PVCs')
+        new_pvc_objs = create_multiple_pvcs(
             sc_name=self.sc_obj.name, namespace=self.namespace,
-            number_of_pvc=self.number_of_pvc
+            number_of_pvc=self.num_of_pvcs
         )
-        log.info(f'Created {self.number_of_pvc} new PVCs.')
-        self.pvc_objs_new = pvc_objs[:]
-
-        # Verify PVCs are Bound
-        for pvc in self.pvc_objs_new:
-            pvc.reload()
-            assert pvc.status == constants.STATUS_BOUND, (
-                f'PVC {pvc.name} is not Bound'
-            )
-        log.info('Verified: Newly created PVCs are in Bound state.')
-
-        # Verify command to delete PVCs
-        ret, out, err = proc.async_communicate()
-        log.info(
-            f'Return values of command: {command}.\nretcode:{ret}\nstdout:'
-            f'{out}\nstderr:{err}'
-        )
-        assert not ret, 'Deletion of PVCs failed'
+        log.info(f'Newly created {self.num_of_pvcs} PVCs are in Bound state.')
+        self.pvc_objs_new.extend(new_pvc_objs)
 
         # Verify PVCs are deleted
-        for pvc in self.pvc_objs_initial:
+        res = pvc_delete.result()
+        assert res, 'Deletion of PVCs failed'
+        log.info('PVC deletion was successful.')
+
+        # Clear pvc_objs list to avoid error in 'create_pvcs' fixture
+        self.pvc_objs.clear()
+
+        # Verify PVCs are deleted
+        for pvc in self.pvc_objs:
             try:
                 pvc.get()
                 return False
@@ -141,15 +87,7 @@ class TestMultiplePvcConcurrentDeletionCreation(ManageTest):
                 assert "not found" in str(exp), (
                     f'Failed to fetch details of PVC {pvc.name}'
                 )
-                log.info(f'Expected: PVC {pvc.name} does not exists ')
-        log.info(f'Successfully deleted initial {self.number_of_pvc} PVCs')
+                log.info(f'Expected: PVC {pvc.name} does not exists')
+        log.info(f'Successfully deleted initial {self.num_of_pvcs} PVCs')
 
-        # Verify PVs using ceph toolbox. PVs should be deleted because
-        # reclaimPolicy is Delete
-        ceph_cmd = f'rbd ls -p {self.cbp_obj.name}'
-        ct_pod = get_ceph_tools_pod()
-        final_pv_list = ct_pod.exec_ceph_cmd(ceph_cmd=ceph_cmd, format='json')
-        assert not any(pv in final_pv_list for pv in self.initial_pvs), (
-            'PVs associated with deleted PVCs still exists'
-        )
-        log.info('Verified: PVs associated with deleted PVCs are also deleted')
+        # TODO: Verify PVs using ceph toolbox. Blocked by Bz 1723656

--- a/tests/manage/pv_services/test_pvc_concurrent_deletion_creation.py
+++ b/tests/manage/pv_services/test_pvc_concurrent_deletion_creation.py
@@ -46,7 +46,7 @@ class TestMultiplePvcConcurrentDeletionCreation(ManageTest):
     """
     Test to verify concurrent creation and deletion of multiple PVCs
     """
-    num_of_pvcs = 10
+    num_of_pvcs = 100
     pvc_size = '3Gi'
 
     def test_multiple_pvc_concurrent_creation_deletion(self):

